src/api_server/api_app.py

```diff
@@ -1,21 +1,73 @@
 import os
 import flask
-from flask import send_file
+from flask import send_file, render_template, request, redirect, url_for, flash
+from werkzeug.utils import secure_filename
 import shutil
 import sys
 
 # get scripts folder to relative path
 sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
-from scripts import sample_script
+#from scripts import sample_script
 
 sys.path.insert(1, '../scripts')
 dir_path = os.path.dirname(os.path.realpath(__file__))
 
 SCRIPT_PATH = dir_path + '/../scripts'
 DATA_FILES_PATH = dir_path + '/../../sample_data'
+TEMPLATE_PATH = dir_path + '/../templates'
+UPLOAD_FOLDER = '/app/static/uploads'
+ALLOWED_EXTENSIONS = {'csv'}
 
-app = flask.Flask(__name__)
+SUCCESS_MSG = ' uploaded successfully! '
 
+#TODO: this might not be enough as not all browsers properly detect file size
+app = flask.Flask(__name__)
+app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER
+app.secret_key = '1u9L#*&I3Ntc'
+app.config['MAX_CONTENT_LENGTH'] = 500 * 1024 * 1024 #500 Megs
+
+def allowed_file(filename):
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+@app.route('/', methods=['GET'])
+def showIndexPage():
+    return render_template('index.html')
+
+#file upload tutorial
+#https://flask.palletsprojects.com/en/1.1.x/patterns/fileuploads/
+@app.route('/', methods=['POST'])
+def uploadCSV():
+    if 'file' not in request.files:
+        flash('no file part', 'error')
+        return redirect(request.url)
+    
+    for file in request.files.getlist('file'):
+        if not allowed_file(file.filename):
+            flash('not a csv: ' + file.filename, 'error')
+            continue
+        try:
+            header = file.stream.readline().decode('utf-8').split(',')[1]
+            cleaned = header.strip()
```

Comment:
I think the name "cleaned" can be more explanatory

Reply:
Agreed and will change to 'column_header'